### PR TITLE
[ADD] project_hr: Link HR with project

### PR DESCRIPTION
**Pull request by pedrobaeza**
Project HR
==========

This is a technical module for adding some HR information to projects and
provide technical fields with HR information:

- Employee categories at project and task level.
- Employee linked to the user assigned to a task.
- Employee categories at user level.

These fields can be used later for extra features like task scheduling or
security restrictions.

It also provides some facilities on project management:

* Limit selection of assigned users according selected employee category at
  task level.
* Limit selection of employee categories at task level depending on the
  selected categories at project level. All are shown if no categories at
  project level.

Configuration
=============

For setting employee categories at project level:

1. Go to *Project > Dashboard*.
2. Click on the 3 vertical dots for unfolding options and select "Settings".
3. Put the wanted employee categories on the field "Employee Categories".

For setting employee categories:

1. Go to *Project > Search > Tasks*.
2. Select or create a new task.
3. Put the wanted employee category on the field "Employee category".
4. If there's already some employee categories selected at project level, those
   will be the only selectable ones in the task.

Usage
=====

1. Go to *Project > Search > Tasks*.
2. If there's an employee category selected in the task, you will only be able
   to select those users whose employee belongs to that category.

Known issues / Roadmap
======================

* Limit project and task visibility according employee categories through
  overriding security methods (as through record rules is not possible without
  modifying existing ones).

### CODE REVIEW

File: project_hr/tests/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from . import test_project_hr
```
Comment (alexey-pelykh) [inline project_hr/tests/__init__.py:1]: Please specify license

**Review by alexey-pelykh — APPROVED** [review]

**Comment by alexey-pelykh** [comment]
@pedrobaeza please enhance code coverage or use `# pragma: no cover` wherever applicable

**Comment by alexey-pelykh** [comment]
@pedrobaeza one more thing - I'm not sure what's the policy about quotes, yet they seem to be inconsistent at the file level

**Comment by pedrobaeza** [comment]
There's no policy established for quotes. As they can be used indifferently, I make use of both.

File: project_hr/tests/test_project_hr.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+from odoo.tests.common import SavepointCase
+from odoo.exceptions import ValidationError
+
+
+class TestProjectHr(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.user = cls.env['res.users'].create({
```
Comment (nikul-serpentcs) [inline project_hr/tests/test_project_hr.py]: Here `cls.env['res.users']`, `cls.env['hr.employee.category']`  use multiple time, use as a global variable

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

File: project_hr/tests/test_project_hr.py (hunk 2 above)
Comment (pedrobaeza) [reply project_hr/tests/test_project_hr.py]: Although not critical, and most as it only appears 2 times, it's done!

**Review by nikul-serpentcs — APPROVED** [review]
Code Review LGTM :+1:

**Pull request merged**